### PR TITLE
Sort items based on price

### PR DESCRIPTION
**Pull request by chiewyx**
- I'm not sure about the exception for invalid sort mode
- Is there a better way to do the optional arguments part?

### CODE REVIEW

**Comment by jingwei55** [comment]
lgtm

File: src/main/java/seedu/duke/command/item/SortItemCommand.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    private boolean isValidMode(String mode) throws InvalidSortModeException {
+        try {
+            if (mode.equals(LOW_HIGH) || mode.equals(HIGH_LOW)) {
+                return true;
+            }
+            throw new InvalidSortModeException(MESSAGE_SORT_MODE_INVALID);
+        } catch (InvalidSortModeException e) {
+            throw new InvalidSortModeException(MESSAGE_SORT_MODE_INVALID);
+        }
```
Comment (bdthanh) [inline src/main/java/seedu/duke/command/item/SortItemCommand.java:80]: try catch statement is unneccessary here

File: src/main/java/seedu/duke/command/item/SortItemCommand.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                sortedItems = itemList.getItemList().stream()
+                        .sorted(comparingDouble(Item::getPricePerDay))
+                        .filter(item -> item.getPricePerDay() >= min && item.getPricePerDay() <= max)
+                        .collect(Collectors.toList());
+            } else if (mode.equals(HIGH_LOW)) {
+                sortedItems = itemList.getItemList().stream()
+                        .sorted(Comparator.comparingDouble(Item::getPricePerDay).reversed())
+                        .filter(item -> item.getPricePerDay() >= min && item.getPricePerDay() <= max)
+                        .collect(Collectors.toList());
+            }
```
Comment (bdthanh) [inline src/main/java/seedu/duke/command/item/SortItemCommand.java:142]: Good use of Stream

File: src/main/java/seedu/duke/command/item/SortItemCommand.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        List<Item> itemsList = sortAndFilter();
+        if (itemsList.size() == 0) {
+            listString.append("There is no items in your filtered list right now");
+        } else {
+            listString.append("Here are ").append(itemsList.size()).append(" item(s) in your filtered list:");
+        }
+        int index = 1;
+        for (Item item : itemsList) {
+            listString.append('\n').append("   ").append(index++).append(". ").append(item.toString(transactionList));
+        }
```
Comment (bdthanh) [inline src/main/java/seedu/duke/command/item/SortItemCommand.java:160]: I think should create a function (formatToPrint(List<Item> listItem)) for this part and call it in the executeCommand

**Review by bdthanh — COMMENTED** [review]
LGTM, can modify some minors part in the future

**Pull request merged**